### DISCUSSION TITLE
Cleaning up of `hydra_oauth2_code` table

### DISCUSSION THREAD

**Discussion by jamiec-w3w** [ory/hydra#3376] [Q&A]
We are using hydra in production and noticed that there is no ability for `janitor` to clean up the `hydra_oauth2_code` table. We are currently at a 117GB table size and about 65 million rows, of which approx 1000 meet the condition `active=1`. Is there a particular reason for `janitor` not being given functionality? And as a remedy - what is the scope if we were to significantly truncate the table? 

We think that users that have asked for an Access Token, but haven't yet exchanged the `authorization code` for said token may have an issue, but we do not think any existing users will be affected - is this correct, or are there side effects we haven't considered?

**Comment by Benehiko** [comment]
Hi @jamiec-w3w 

A lot has changed with Hydra v2 in terms of the database tables and I believe it should be possible to cascade delete this as well :thinking:.

Please take a look at this discussion https://github.com/ory/hydra/discussions/2556

We also have an issue regarding this https://github.com/ory/hydra/issues/2514 and I think since v2 has been released this should be easier to implement.

wdyt? @aeneasr
